org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/ui/util/IResourcesSetupUtil.java

```diff
@@ -220,7 +221,11 @@ protected void execute(IProgressMonitor monitor)
 					InterruptedException {
 				create(file.getParent());
 				file.delete(true, monitor());
-				file.create(new StringInputStream(s), true, monitor());
+				try {
+					file.create(new StringInputStream(s, file.getCharset(true)), true, monitor());
+				} catch (UnsupportedEncodingException exc) {
+					throw new IllegalArgumentException(exc.getMessage(), exc);
```

Comment:
I think it is not correct to throw IAE here, since the exception is not caused by an argument. The UEE is thrown when the determined encoding for the file is not supported. A CoreException is the right alternative here.